converter/src/DelphesEDM4HepConverter.cc

```diff
@@ -414,6 +414,20 @@ std::optional<edm4hep::ReconstructedParticle> DelphesEDM4HepConverter::getMatchi
   return {};
 }
 
+edm4hep::MCRecoParticleAssociationCollection* DelphesEDM4HepConverter::createExternalRecoAssociations(std::unordered_map<UInt_t, edm4hep::ConstMCParticle> mc_map) {
+
+  auto mcRecoRelations = new edm4hep::MCRecoParticleAssociationCollection();
+    for (auto particleID: mc_map) {
```

Comment:
```suggestion
edm4hep::MCRecoParticleAssociationCollection* DelphesEDM4HepConverter::createExternalRecoAssociations(const std::unordered_map<UInt_t, edm4hep::ConstMCParticle>& mc_map) {

  auto mcRecoRelations = new edm4hep::MCRecoParticleAssociationCollection();
    for (const auto& particleID: mc_map) {
```
Try to pass by const ref (if possible with the iteration)